lib/active_model/serializer.rb

```diff
@@ -62,13 +62,27 @@ class << self
     def self.serializer_lookup_chain_for(klass)
       chain = []
 
-      resource_class_name = klass.name.demodulize
-      resource_namespace = klass.name.deconstantize
+      resource_class_name   = klass.name.demodulize
+      resource_namespace    = klass.name.deconstantize
       serializer_class_name = "#{resource_class_name}Serializer"
 
-      chain.push("#{name}::#{serializer_class_name}") if self != ActiveModel::Serializer
-      chain.push("#{resource_namespace}::#{serializer_class_name}")
+      tmp_namespace   = name.to_s
+      name_namespaces = [tmp_namespace]
+
+      until tmp_namespace.empty?
```

Comment:
do you need to have two loops here? I imagine you could add to the chain array as you deconstantized the namespaces.

Could you also add some documentation (comments) in here explaining how the chain array looks after different steps of this process?